module/zfs/zio.c

```diff
@@ -3217,14 +3224,40 @@ zio_write_gang_block(zio_t *pio, metaslab_class_t *mc)
 		memset(zp.zp_iv, 0, ZIO_DATA_IV_LEN);
 		memset(zp.zp_mac, 0, ZIO_DATA_MAC_LEN);
 
-		zio_t *cio = zio_write(zio, spa, txg, &gbh->zg_blkptr[g],
-		    has_data ? abd_get_offset(pio->io_abd, pio->io_size -
-		    resid) : NULL, psize, psize, &zp,
-		    zio_write_gang_member_ready, NULL,
+		uint64_t min_size = zio_roundup_alloc_size(spa,
+		    resid / (SPA_GBH_NBLKPTRS - g));
+		min_size = MIN(min_size, resid);
+		IMPLY(min_size < spa->spa_min_alloc, min_size == resid);
+		IMPLY(min_size >= spa->spa_min_alloc, min_size <= resid);
```

Comment:
The second `IMPLY()` is always true, since we just calculated `MIN()`.  The usefulness of the first is also questionable IMHO, it just bloats the code.